Bugfix/850/resolve es lint error

am-Anmol opened the pull request:
Target #850 
## Summary
1. Most of the errors solved with command `./node_modules/.bin/eslint resources/js/ Modules/*/Resources/assets/js/ --ext .js,.vue --fix`.
2. Rest were solved manually solved.

rathorevaibhav commented on Modules/Prospect/Resources/assets/js/components/ProspectProgressItem.vue:
```diff
@@ -34,36 +34,36 @@
-                state:'view'
-            }
-        },
+new Vue({
```
isn't this supposed to be exported? How is the main js file able to import this otherwise?

am-Anmol replied:
Yes, there should be exported, previously interpreted wrongly.

rathorevaibhav commented on Modules/User/Resources/assets/js/components/RoleListing.vue:
```diff
@@ -27,40 +27,40 @@
 <script>
-    export default {
-        props:[ 'roles', 'updateRoute', 'permissions'],
+export default {
```
why not `new Vue` here?

am-Anmol replied:
There should be `exported` only, in the previous file using the `new vue` but there should also be `exported`.

rathorevaibhav commented on Modules/User/Resources/assets/js/components/UpdateRolePermissionModal.vue:
```diff
@@ -36,62 +36,62 @@
-                permissions.forEach((permission) => this.permissionInputs[permission.id].checked =  true );
-            },
+			let permissionInputContainer = document.querySelector("#update_role_permissions_modal");
+			let allPermissionInputs = permissionInputContainer.querySelectorAll("input[type=\"checkbox\"]");
```
```suggestion
			let allPermissionInputs = permissionInputContainer.querySelectorAll("input[type='checkbox']");
```

am-Anmol replied:
Ok, I make these changes.

rathorevaibhav commented on Modules/User/Resources/assets/js/components/UpdateUserRoleModal.vue:
```diff
@@ -59,110 +59,110 @@
+
+
+			let roleInputContainer = document.querySelector("#update_user_roles_modal");
+			let allRoleInputs = roleInputContainer.querySelectorAll("input[type=\"checkbox\"]");
```
```suggestion
			let allRoleInputs = roleInputContainer.querySelectorAll("input[type='checkbox']");
```

am-Anmol replied:
Ok, I make these changes.

rathorevaibhav commented on resources/js/bootstrap.js:
```diff
@@ -19,22 +19,22 @@ try {
  */
 
-let token = document.head.querySelector('meta[name="csrf-token"]');
+let token = document.head.querySelector("meta[name=\"csrf-token\"]");
```
```suggestion
let token = document.head.querySelector("meta[name='csrf-token']");
```

am-Anmol replied:
Ok, I make these changes.

rathorevaibhav commented on resources/js/components/InvoiceProjectComponent.vue:
```diff
@@ -3,19 +3,19 @@
             <label>Project</label>
             <select class="form-control" v-model="projectId" v-on:change="updateActiveProject()">
-                <option v-for="project in projects" :value="project.id">{{ project.name }}</option>
+                <option v-for="project in projects" :key="project.id">{{ project.name }}</option>
```
Is `key` same as `value`? My understanding is that `value` is used for form submission. That's what goes into the database. Have you tested the functionality?

am-Anmol replied:
No `key` and `value` are different. Yes, `value` is used for form submission and `key` is used in v-for vuejs to have direct reference to those items
In version 2.2.0+, when using v-for with a component, a key is now required.

rathorevaibhav commented on resources/js/components/ProjectStageComponent.vue:
```diff
@@ -55,7 +55,7 @@
                                     <label for="name" class="field-required">Type</label>
                                     <select name="type" id="type" class="form-control" required="required" v-model="stageType">
-                                        <option v-for="(displayName, type) in configs.projectTypes" :value="type">
+                                        <option v-for="(displayName, type) in configs.projectTypes" :key="type">
```
same comment as before. Please check the difference between `key` and `value`.

am-Anmol replied:
Answered above https://github.com/ColoredCow/portal/pull/860#discussion_r685712700

rathorevaibhav commented on resources/js/components/ProjectStageComponent.vue:
```diff
@@ -178,129 +178,129 @@
-                    billing: billings,
-                    new_billing: newBillings,
-                });
+new Vue( {
```
same comment as above. Is the main js file is importing this component without issues when you've removed the `export default`?

am-Anmol replied:
There should be `exported` only, in the previous file using the `new vue` but there should also be `exported`.

rathorevaibhav commented on Modules/Prospect/Resources/assets/js/components/ProspectHistory.vue:
```diff
@@ -25,31 +25,31 @@
+	},
+
+	mounted: function() {
+		this.$comments = this.$bookComments.reverse();
```
can you explain a bit about this change? Why is this needed? What was the error you were trying to resolve?

My question arises because expect for the mounted function, you don't need dollar sign to access this data.

am-Anmol replied:
Yes, there is no need to using `$` with `this`, previously interpreted in the wrong way.

am-Anmol commented:
## Summary
1. Added requested changes given by @rathorevaibhav.
2. Removed manual errors solved because it is not correct as this requires more understanding and research of vue js.

All errors solved by the command are resolved and committed in this pull request, for all left manual errors creating a new pull request soon with the correct solution.

P4NK4J commented:
@am-Anmol I see some of the linting checks are being failed in the PR. I think let's get them fixed first, then it would be easier to look over the PR.

rathorevaibhav commented:
@P4NK4J the linting checks are failing in all PRs and master branch. This PR along with #849 #850 #851 targets to get these linting issues fixed.

If we can get an update from @am-Anmol on how much he has fixed here, that'd be better. I think we can let the checks fail for now.

rathorevaibhav commented:
@ColoredCow/team-portal-reviewers please take it forward for review.

P4NK4J commented:
@am-Anmol , Could you please resolve the feedback which @rathorevaibhav gave on the previous review, as of now it's not clear if the questions are answered or the suggested changes are made.

rathorevaibhav commented on Modules/Prospect/Resources/assets/js/components/ProspectHistory.vue:
```diff
@@ -25,31 +25,31 @@
+	},
+
+	mounted() {
+		this.comments = this.bookComments.slice().reverse();
```
why is `slice()` needed?

am-Anmol replied:
Actually, it's showing an error `mutation of 'bookComments' props`, then tried below code:
```
mounted() {
this.comments = this.reverseComment();
 },
methods: {
 async reverseComment() {
 return this.bookComments.reverse();
 }
```
then the same error occurs in this line  `return this.bookComments.reverse()` then I try without `reverse()` then it working fine, then I found the error caused due to `reverse()`, then referred [this](https://renatello.com/vue-js-reverse-array/#:~:text=The%20easiest%20solution,li%3E%0A%20%20%3C/ol%3E%0A%3C/div%3E) for how to reverse in vue js. then by using `slice` its resolves the error.

rathorevaibhav commented on Modules/Prospect/Resources/assets/js/components/ProspectProgressItem.vue:
```diff
@@ -18,7 +18,7 @@
             <div>
                 <div v-if="state == 'edit'">
-                    <textarea v-model="comment.body" class="form-control"> </textarea>
+                    <textarea v-model="body" class="form-control"> </textarea>
```
I don't understand this change. Can you elaborate what was happening before?

am-Anmol replied:
The error that occurs here is `mutation of 'comment' props`, so for solving I referred from [here](https://dev.to/nop33/fixing-no-mutating-props-vue-error-with-v-model-54oh#:~:text=-%20%3Cv-text-field%20v-model%3D%22item.name%22%20label%3D%22Item%20name%22%20/%3E%0A%2B%20%3Cv-text-field%20v-model%3D%22name%22%20label%3D%22Item%20name%22%20/%3E), how to avoid mutating-props Vue Error with v-model.

rathorevaibhav commented on resources/js/components/ProjectStageComponent.vue:
```diff
@@ -55,18 +55,18 @@
                                 <div class="form-group col-md-8">
                                     <label for="end_date">End date</label>
-                                    <input type="date" class="form-control" name="end_date" id="end_date" placeholder="dd/mm/yy" v-model="stage.end_date">
+                                    <input type="date" class="form-control" name="end_date" id="end_date" placeholder="dd/mm/yy" v-model="end_date">
```
as my understanding is with `v-model`, it binds the input value to an existing variable (from [this article](https://dev.to/vue-storefront/vue-during-coffee-break-using-v-model-with-custom-components-3bo9#:~:text=Understanding%20v%2Dmodel,%2C%20select%20%2C%20textarea%20and%20radio%20.))

But I don't see any variable named `name`, `start_date` or `end_date`. Can you elaborate a bit more on this change? Maybe my understanding is incorrect.

am-Anmol replied:
For here, the same error occurs `mutation of 'stage' props`, then referred from above document [here](https://dev.to/nop33/fixing-no-mutating-props-vue-error-with-v-model-54oh#:~:text=-%20%3Cv-text-field%20v-model%3D%22item.name%22%20label%3D%22Item%20name%22%20/%3E%0A%2B%20%3Cv-text-field%20v-model%3D%22name%22%20label%3D%22Item%20name%22%20/%3E).

rathorevaibhav reviewed (changes requested):
@am-Anmol overall great work. Just some final questions. Please help me understand before making code changes.

GauravGusain98 commented on Modules/User/Resources/assets/js/components/UpdateUserRoleModal.vue:
```diff
@@ -59,110 +59,110 @@
+			return this.config.website_url
+                        + "/wp/wp-admin/user-edit.php"
+                        + "?idp_referrer=" + window.location.href
+                        + "&user_id=" +  this.user.websiteUser.ID;
```
```suggestion
                        + "&user_id=" + this.user.websiteUser.ID;
```

am-Anmol replied:
Ok, I make these changes.

GauravGusain98 commented on resources/js/components/Book/ProspectProgress.vue:
```diff
@@ -25,31 +25,31 @@
+	},
+
+	mounted: function() {
+		this.$comments = this.$bookComments.reverse();
```
Why you used `$` is used after `this`. Is it even working?

am-Anmol replied:
I forgot to make these changes, there is no need to using `$` with `this`.
I have made changes in the `Modules/Prospect/Resources/assets/js/components/ProspectHistory.vue` , like same changes required on both. 
For more clarity on why I make these changes you can check out here: https://github.com/ColoredCow/portal/pull/860#discussion_r684371749

GauravGusain98 commented on resources/js/components/ProjectDetailsComponent.vue:
```diff
@@ -23,13 +23,13 @@
                                 <p><strong>Contribution</strong></p>
                                 <ul class="unstyled-list p-0">
-                                    <li v-for="employee in project.employees">0</li>
+                                    <li v-for="employee in project.employees" :key="employee.id" >0</li>
```
I am seeing this change in many places. 
`project.employees" :key="employee.id"` 

Is it required due to some package upgrade and did you test it in your local?

am-Anmol replied:
Yes, it's due to the `vue js version` update, In `v 2.2.0+`, when using `v-for` with a component, a `key` is now required.
Mentioned some `articles` for your better understanding:
->[Stackoverflow](https://stackoverflow.com/questions/47395535/why-is-the-key-attribute-needed-in-vuejs-v-for)
->[Vuejs Offical list rendering](https://vuejs.org/v2/guide/list.html)
->[Understanding of key attribute in vue js](https://www.telerik.com/blogs/in-vue-when-do-i-actually-need-the-key-attribute-and-why#:~:text=should%20you%20care%3F-,Understanding%20the%20Basics%20of%20%3Akey%20%F0%9F%94%91,Object%20Constancy,-Animations%20are%20not)

GauravGusain98 reviewed (changes requested):
Left some comments. Please have a look.

P4NK4J commented on resources/js/components/Dashboard/UserDashboardProjects.vue:
```diff
@@ -12,29 +12,29 @@
-        mounted() {
-            //this.getReadBooks();
-        }
+	mounted() {
+		//this.getReadBooks();
+	}
```
What's the use of commented code here?

am-Anmol replied:
This code is previously commented on before I started working on this task.

P4NK4J replied:
Let's remove that if not being used, and merge the PR

P4NK4J replied:
I meant to remove the commented code earlier, just to confirm by removing `mounted()` is everything working fine on the local machine?

am-Anmol replied:
Yes, it's working fine without using `mounted()`.

P4NK4J reviewed (approved):
Minor comment, else looking good.
